### DISCUSSION TITLE
Breaking changes for micromamba v2

### DISCUSSION THREAD

**Discussion by wholtz** [mamba-org/micromamba-docker#396] [General]
An alpha release of `micromamba` v2.0 dropped today ([release notes](https://github.com/mamba-org/mamba/releases/tag/2023.12.14)]. The upcoming release of v2.0 would be a good time to make any breaking changes to the docker images. I have two potential breaking changes to discuss:

1. Back in the very early days of this image, I set the working directory to `/tmp`.  If the working directory is not set in an image, it defaults to `/`. I don't want to use `/` as we default to running as a non-root user and `/` is not writable as that user. The user should be able to write into the working directory. The `/home/mambuser` would be another option, but that gets weird if the user gets changed at runtime (`docker run -u my_user:my_group`). Some official docker images put you into an application specific directory (containers such as `tomcat`, `httpd`,`wordpress`) and this might be the right approach for us. Right now `/opt/conda`  (or maybe `/opt/micromamba`) is sounding like the right approach to me. 
2. Our tags could be more consistent and easier to use. (Alpine is just broken and I need to fix that #379). I feel we should get away from using release names in tags, such as 'bookworm', 'jammy',  and use more numbers (which may be years and month values). See #249. 

I'm interested in hearing thoughts on the above changes or any other areas where you think breaking changes might be worth considering.

**Comment by maresb** [comment]
My 2¢:

1. I've become quite fond of the `/tmp/` default working directory. The reason is that for virtually all purposes it's not broken, making it great for prototyping, but it's simultaneously awkward enough to encourage users to set it explicitly rather than mindlessly following the default. I personally wouldn't change it, but it's not something I feel strongly about.

2. Agreed, I really dislike release names because I can never remember their corresponding dates.

**Reply by wholtz** [reply]
Since we are split over the working directory and it isn't broken, we should stay with the status quo.
